synapse/handlers/e2e_keys.py

```diff
@@ -259,34 +292,60 @@ def claim_client_keys(destination):
             ),
         )
 
+        opentracing.log_kv({"one_time_keys": json_result, "failures": failures})
         return {"one_time_keys": json_result, "failures": failures}
 
     @defer.inlineCallbacks
+    @opentracing.tag_args
     def upload_keys_for_user(self, user_id, device_id, keys):
+
         time_now = self.clock.time_msec()
 
         # TODO: Validate the JSON to make sure it has the right keys.
         device_keys = keys.get("device_keys", None)
+        opentracing.set_tag("device_keys", device_keys)
         if device_keys:
             logger.info(
                 "Updating device_keys for device %r for user %s at %d",
                 device_id,
                 user_id,
                 time_now,
             )
+            opentracing.log_kv(
+                {
+                    "message": "Updating device_keys for user.",
+                    "user_id": user_id,
```

Comment:
don't we get these already from the tags? (likewise several times below)

Reply:
Yes we do, I just thought it would be nice to have all the relevant information in the log message which is recommended by the structured logging approach.